ocaml-lsp-server/src/hover_req.ml

```diff
@@ -148,36 +140,109 @@ let hover_at_cursor parsetree (`Logical (cursor_line, cursor_col)) =
     (if is_at_cursor mtyp.pmty_loc then
        match mtyp.pmty_desc with
        | Pmty_ident { loc; _ } ->
-         if is_at_cursor loc then result := Some `Type_enclosing
+         if is_at_cursor loc then result := Some Type_enclosing
        | _ -> ());
     Ast_iterator.default_iterator.module_type self mtyp
   in
   (* Hover structure items *)
   let structure_item (self : Ast_iterator.iterator)
       (item : Parsetree.structure_item) =
     match item.pstr_desc with
+    | Pstr_type (_, tds) ->
+      List.iter
+        ~f:(fun (td : Parsetree.type_declaration) ->
+          let deriving_attr =
+            List.find
+              td.ptype_attributes
+              ~f:(fun (attr : Parsetree.attribute) ->
+                is_at_cursor attr.attr_loc && is_deriving_attr attr)
+          in
+          match deriving_attr with
+          | Some attr -> result := Some (Ppx_str_attr (item, attr.attr_loc))
+          | None -> result := Some Type_enclosing)
+        tds
     | Pstr_module desc when is_at_cursor desc.pmb_name.loc ->
-      result := Some `Type_enclosing
+      result := Some Type_enclosing
+    | Pstr_modtype mtd -> (
+      let deriving_attr =
+        List.find mtd.pmtd_attributes ~f:(fun (attr : Parsetree.attribute) ->
+            is_at_cursor attr.attr_loc && is_deriving_attr attr)
+      in
+      match deriving_attr with
+      | Some attr -> result := Some (Ppx_str_attr (item, attr.attr_loc))
+      | None -> result := Some Type_enclosing)
+    | Pstr_exception tc -> (
+      let deriving_attr =
+        List.find tc.ptyexn_attributes ~f:(fun (attr : Parsetree.attribute) ->
+            is_at_cursor attr.attr_loc && is_deriving_attr attr)
+      in
+      match deriving_attr with
+      | Some attr -> result := Some (Ppx_str_attr (item, attr.attr_loc))
+      | None -> result := Some Type_enclosing)
+    | Pstr_typext tex -> (
+      let deriving_attr =
+        List.find tex.ptyext_attributes ~f:(fun (attr : Parsetree.attribute) ->
+            is_at_cursor attr.attr_loc && is_deriving_attr attr)
+      in
+      match deriving_attr with
+      | Some attr -> result := Some (Ppx_str_attr (item, attr.attr_loc))
+      | None -> result := Some Type_enclosing)
```

Comment:
So all these cases were not handled by the previous implementation ?

Reply:
Yes previously, only a ppx deriving attribute on a type_declaration will work